fix: pr template not being used and make some changes (#15893)
immich-app/immich: .github/pull_request_template.md

## Description

<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

Fixes # (issue)

## How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- [ ] Test A
- [ ] Test B

> <details><summary><h2>Screenshots (if appropriate)</h2></summary>

Author: Made this a dropdown so that images won't take up the entire screen.

<!-- Images go below this line. -->

</details>

<!-- API endpoint changes (if relevant)
## API Changes
The `/api/something` endpoint is now `/api/something-else`
-->

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if applicable
- [ ] I have no unrelated changes in the PR.
- [ ] I have confirmed that any new dependencies are strictly necessary.

> - [ ] I have written unit tests for new code (if applicable)

Reviewer: Perhaps?

```suggestion
- [ ] I have written tests for new code (if applicable)
```

- [ ] I have followed naming conventions/patterns in the surrounding code

> - [ ] All code in `domain/` uses repositories for implementations of database calls, filesystem operations, etc.

Reviewer 2: ```suggestion
- [ ] All code in `domain/` uses repositories for implementations of database calls, filesystem operations, etc.
```
```suggestion
- [ ] All code in `src/services` use repositories implementations for database calls, filesystem operations, etc.
```

> - [ ] All code in `infra/` is pretty basic/simple and does not have any immich specific logic (that belongs in domain)

Author: Added jrasm91 checklist items from https://discord.com/channels/979116623879368755/1071165397228855327/1335978945031442442

> - [ ] All code in `infra/` is pretty basic/simple and does not have any immich specific logic (that belongs in domain)

Reviewer 2: ```suggestion
- [ ] All code in `src/repositories/` is pretty basic/simple and does not have any immich specific logic (that belongs in `src/services`)
```
